src/data/actions/index.test.js

```diff
@@ -1,65 +1,104 @@
-// import configureMockStore from 'redux-mock-store';
-// import thunk from 'redux-thunk';
-// import fetchMock from 'fetch-mock';
+import configureMockStore from 'redux-mock-store';
+import thunk from 'redux-thunk';
+import fetchMock from 'fetch-mock';
 
-// import {
-//   startedFetchingPosts,
-//   finishedFetchingPosts,
-//   getPosts,
-//   fetchPosts,
-// } from './index';
-// import {
-//   STARTED_FETCHING_POSTS,
-//   GET_POSTS,
-//   FINISHED_FETCHING_POSTS,
-// } from '../constants/ActionType';
+import { fetchCourseOutline } from './index';
+import {
+  STARTED_FETCHING_COURSE_OUTLINE,
+  FINISHED_FETCHING_COURSE_OUTLINE,
+  GET_COURSE_OUTLINE,
+} from '../constants/ActionType';
 
-// const mockStore = configureMockStore([thunk]);
+const mockStore = configureMockStore([thunk]);
 
-// describe('actions', () => {
-//   afterEach(() => {
-//     fetchMock.reset();
-//     fetchMock.restore();
-//   });
+describe('actions', () => {
+  afterEach(() => {
+    fetchMock.reset();
+    fetchMock.restore();
+  });
 
-//   it('sends started fetching post action', () => {
-//     const expected = { type: STARTED_FETCHING_POSTS };
-//     expect(startedFetchingPosts()).toEqual(expected);
-//   });
+  describe('fetchCourseOutline', () => {
+    it('fetches outline and converts flat map to nested tree', () => {
+      const courseId = 'course-123 demo test course';
+      const responseOutline = {
+        root: 'course-123',
+        blocks: {
+          'course-123': {
+            id: 'course-123',
+            display_name: 'Root Node',
+            student_view_url: 'http://www.example.com/',
+            type: 'Course',
+            descendants: ['chapter-1', 'chapter-2'],
+          },
+          'chapter-1': {
+            id: 'chapter-1',
+            display_name: 'Chapter 1',
+            student_view_url: 'http://www.example.com/chapter1',
+            type: 'Chapter',
+            descendants: ['section-1'],
+          },
+          'chapter-2': {
+            id: 'chapter-2',
+            display_name: 'Chapter 2',
+            student_view_url: 'http://www.example.com/chapter2',
+            type: 'Chapter',
+            descendants: [],
+          },
+          'section-1': {
+            id: 'section-1',
+            display_name: 'Section 1',
+            student_view_url: 'http://www.example.com/chapter1/section2',
+            type: 'Section',
+            descendants: [],
+          },
+        },
+      };
+      fetchMock.getOnce(`http://localhost:18000/api/courses/v1/blocks/?course_id=${encodeURIComponent(courseId)}&username=staff&depth=all&nav_depth=3&block_types_filter=course,chapter,sequential,vertical`, {
```

Comment:
nit: For long URLs with params like this, it might be cleaner to use something like [`querystring`](https://nodejs.org/api/querystring.html#querystring_querystring_stringify_obj_sep_eq_options) to represent the params as an object instead (also would apply to the LMS API request in the `CourseOutline` action).

Reply:
👍 This will eventually (soon) move into a constant, lets make sure to use `querystring` when it does